src/System.Device.Gpio/System/Device/Gpio/Drivers/HummingBoardDriver.cs

```diff
@@ -7,7 +7,7 @@ namespace System.Device.Gpio.Drivers
     /// <summary>
     /// A GPIO driver for the HummingBoard.
     /// </summary>
-    public partial class HummingBoardDriver // Different base classes declared in HummingboardDriver.Linux.cs and HummingboardDriver.Windows.cs
+    public class HummingBoardDriver : Windows10Driver

```

Comment:
This change I don't fully like here, it is definitely less than ideal to have HummingBoardDriver which could be running in Linux to have a base class called Windows10Driver. We should either think on how to either make the HummingBoardDriver extend GpioDriver in Windows as well, or to rename Windows10Driver so that we can make something less platform-named. Both changes would be breaking.

Reply:
The reason for changing it that way was that the HummingBoardDriver.Linux.cs implementation file previously consisted of only "NotImplementedException" methods, so it newer worked. Therefore I thought it was anyway only supported on Windows. 

If this was ever supported on linux, we could just provide a HummingBoardLinuxDriver (and move the magic of choosing the right one to the HummingBoard class or at least the GetBestDriverForBoard() method).

Reply:
having it never worked in the past is different as have it never work at all in the future. Imagine we eventually decide to add a HummingBoard driver in the future for Linux, and if so and we keep this as is, it means we would have to have an implementation of Windows10Driver on Linux as well so that we don't introduce a breaking change. This is obviosly less than ideal. In hindsight, we probably should have never exposed drivers with platforms name on them, it would be better to have a "GenericDriver" or "PlatformDriver" and have different implementations for it (what today is Wiindows10Driver and UnixDriver) but I guess that boat has sailed and will brings us some trouble now that we are unifying things.

Reply:
We could make the HummingBoardDriver aggregate its underlying driver (UnixDriver or Windows10Driver) instead of inheriting it. That would make the actual implementation flexible, and, despite it technically being a breaking change, probably nobody would really notice.